tests/integration/devfile/cmd_devfile_list_test.go

```diff
@@ -29,79 +24,110 @@ var _ = Describe("odo list with devfile", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	When("a component created in 'app' application", func() {
+	When("a component created for deployment", func() {
 
 		BeforeEach(func() {
-			helper.Cmd("odo", "create", "--project", commonVar.Project, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
+			helper.Chdir(commonVar.Context)
 		})
 
-		It("should show the component as 'Not Pushed'", func() {
-			output := helper.Cmd("odo", "list").ShouldPass().Out()
-			Expect(helper.Suffocate(output)).To(ContainSubstring(helper.Suffocate(fmt.Sprintf("%s%s%s%sNotPushed", "app", cmpName, commonVar.Project, "nodejs"))))
+		AfterEach(func() {
+			helper.Chdir(commonVar.Context)
+			helper.Cmd("odo", "v2delete", "-a").ShouldPass()
 		})
 
-		It("should show the component as 'Not Pushed' in JSON output", func() {
-			output := helper.Cmd("odo", "list", "-o", "json").ShouldPass().Out()
-			values := gjson.GetMany(output, "kind", "devfileComponents.0.kind", "devfileComponents.0.metadata.name", "devfileComponents.0.status.state")
-			expected := []string{"List", "Component", cmpName, "Not Pushed"}
-			Expect(helper.GjsonExactMatcher(values, expected)).To(Equal(true))
-		})
-	})
-	Context("devfile has missing metadata", func() {
-		// Note: We will be using SpringBoot example here because it helps to distinguish between language and projectType.
-		// In terms of SpringBoot, spring is the projectType and java is the language; see https://github.com/redhat-developer/odo/issues/4815
+		It("show an odo deploy in the list", func() {
 
-		var metadata devfilepkg.DevfileMetadata
+			By("should display the component as 'Not Pushed' in odo list", func() {
+				stdOut := helper.Cmd("odo", "list").ShouldPass().Out()
+				Expect(stdOut).To(ContainSubstring("Not Pushed"))
+			})
 
-		// checkList checks the list output (both normal and json) to see if it contains the expected componentType
-		var checkList = func(componentType string) {
-			By("checking the normal output", func() {
-				stdOut := helper.Cmd("odo", "list", "--context", commonVar.Context).ShouldPass().Out()
-				Expect(stdOut).To(ContainSubstring(componentType))
+			// Fake the odo deploy image build / push passing in "echo" to PODMAN
+			stdout := helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
+			By("building and pushing image to registry", func() {
+				Expect(stdout).To(ContainSubstring("build -t quay.io/unknown-account/myimage -f " + filepath.Join(commonVar.Context, "Dockerfile ") + commonVar.Context))
+				Expect(stdout).To(ContainSubstring("push quay.io/unknown-account/myimage"))
```

Comment:
I don't think we need to check this in this file. Ensuring that `odo deploy` passes should be fine IMO.

Reply:
I wanted to keep this in to ensure that `odo deploy` still works / shows the commands and we don't get a false negative on the testing suite. But I'll remove it!